### PR TITLE
Cluster-scope-like behavior, autoinstall of KnativeEventingKafka as knative-eventing/knative-eventing-kafka

### PR DESCRIPTION
**Pull request by syedriko**

### CODE REVIEW

**Comment by syedriko** [comment]
A port of https://github.com/openshift-knative/knative-eventing-operator/pull/45 to the Kafka operator

**Comment by syedriko** [comment]
Do not merge just yet

File: deploy/olm-catalog/knative-kafka-operator/0.6.0/knative-kafka-operator.v0.6.0.clusterserviceversion.yaml
Comment (matzew) [inline deploy/olm-catalog/knative-kafka-operator/0.6.0/knative-kafka-operator.v0.6.0.clusterserviceversion.yaml:161]: this means, you need to update the PR 420 on the community-operator repo :-)

**Comment by matzew** [comment]
@syedriko let's add something along the lines of https://github.com/openshift-knative/knative-eventing-operator/commit/d6c59b4b237a1f616527fd6bb31e6796ef712823   to give some clarification to our users

**Comment by matzew** [comment]
If @jcrossley3 has no concerns - :shipit:

**Review by matzew — APPROVED** [review]
see comments ...

File: deploy/olm-catalog/knative-kafka-operator/0.6.0/knative-kafka-operator.v0.6.0.clusterserviceversion.yaml
Comment (syedriko) [reply deploy/olm-catalog/knative-kafka-operator/0.6.0/knative-kafka-operator.v0.6.0.clusterserviceversion.yaml:161]: Absolutely, as soon as this PR merges

File: pkg/controller/knativeeventingkafka/knativeeventingkafka_controller.go
Comment (jcrossley3) [inline pkg/controller/knativeeventingkafka/knativeeventingkafka_controller.go:28]: Sharing a ns with another component _may_ be a problem. But if it is, I'd prefer to fix manifestival than force you to use your own ns. Just be aware that if an operator creates a ns, it will probably try to delete that ns when the associated CR is deleted. If the ns was already there, it won't.

**Review by jcrossley3 — COMMENTED** [review]
Why so many vendor changes on this?

**Comment by syedriko** [comment]
> Why so many vendor changes on this?

That's what `dep ensure -update` pulled in...

**Comment by jcrossley3** [comment]
> > Why so many vendor changes on this?
> 
> That's what `dep ensure -update` pulled in...

You gotta be careful with that. I typically only use that to update a specific dep. In the future, whenever you feel the need to update *everything*, split that out into its own commit/pr, just to facilitate rolling back if need be.

**Comment by syedriko** [comment]
> > > Why so many vendor changes on this?
> > 
> > 
> > That's what `dep ensure -update` pulled in...
> 
> You gotta be careful with that. I typically only use that to update a specific dep. In the future, whenever you feel the need to update _everything_, split that out into its own commit/pr, just to facilitate rolling back if need be.

Fixed, only manifestival is updated.

**Review by jcrossley3 — APPROVED** [review]
Perfect, thanks!
/lgtm

File: pkg/controller/knativeeventingkafka/knativeeventingkafka_controller.go
Comment (syedriko) [reply pkg/controller/knativeeventingkafka/knativeeventingkafka_controller.go:28]: This latest iteration of the Kafka operator places all the ns-scoped k8s objects it creates into the knative-eventing namespace, which the code assumes to exist, since the Kafka objects depend on Eventing being installed [in the knative-eventing namespace]. So I think we're in the clear wrt to the Kafka operator trying to remove the knative-eventing namespace.

**Comment by syedriko** [comment]
> @syedriko let's add something along the lines of [openshift-knative/knative-eventing-operator@d6c59b4](https://github.com/openshift-knative/knative-eventing-operator/commit/d6c59b4b237a1f616527fd6bb31e6796ef712823) to give some clarification to our users

Done

**Pull request merged**
